gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/jupiter/core/v4/endpoint/loadbalancer/WeightedRandomLoadBalancer.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.jupiter.core.v4.endpoint.loadbalancer;
+
+import io.gravitee.gateway.jupiter.core.v4.endpoint.ManagedEndpoint;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * This loadbalancer will randomly choose an endpoint based on total remaining weigh and verify that its remaining weigh is higher than 0, otherwise it will pick the next one.
```

Comment:
```suggestion
 * This loadbalancer will randomly choose an endpoint based on total remaining weight and verify that its remaining weight is higher than 0, otherwise it will pick the next one.
```